pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/IntermediateResultsBlock.java

```diff
@@ -360,6 +360,9 @@ private void setDataTableColumn(ColumnDataType columnDataType, DataTableBuilder
       case DOUBLE:
         dataTableBuilder.setColumn(columnIndex, (double) value);
         break;
+      case BIG_DECIMAL:
+        dataTableBuilder.setColumn(columnIndex, value);
```

Comment:
We should add a `setColumn(int colId, BigDecimal value)` into the `DataTableBuilder` instead of reusing the one for general objects.